bin/plugin/commands/packages.js

```diff
@@ -349,171 +380,174 @@ async function publishPackagesToNpm(
- * @param {string}      [confirmationMessage] Confirmation message to show at first.
+ * @param {WPPackagesConfig} config         Command config.
+ * @param {string[]}         customMessages Custom messages to print in the terminal.
  *
  * @return {Promise<Object>} GitHub release object.
  */
-async function prepareForPackageRelease(
-	releaseType,
-	minimumVersionBump = 'patch',
-	confirmationMessage = 'Ready to go?'
-) {
-	await askForConfirmation( confirmationMessage );
-
-	// Cloning the Git repository.
-	const abortMessage = 'Aborting!';
-	const gitWorkingDirectoryPath = await runGitRepositoryCloneStep(
-		abortMessage
+async function runPackagesRelease( config, customMessages ) {
+	log(
+		formats.title(
+			'\n💃 Time to publish WordPress packages to npm 🕺\n\n'
+		),
+		"To perform a release you'll have to be a member of the WordPress Team on npm.\n",
+		...customMessages
 	);
+
 	const temporaryFolders = [];
-	temporaryFolders.push( gitWorkingDirectoryPath );
+	if ( config.interactive ) {
+		await askForConfirmation( 'Ready to go?' );
 
-	// Checking out the WordPress release branch and doing sync with the last plugin release.
-	const { releaseBranch } = await runWordPressReleaseBranchSyncStep(
-		gitWorkingDirectoryPath,
-		releaseType,
-		abortMessage
-	);
+		// Cloning the Git repository.
+		config.gitWorkingDirectoryPath = await runGitRepositoryCloneStep(
+			config.abortMessage
+		);
+		temporaryFolders.push( config.gitWorkingDirectoryPath );
+	}
 
-	const commitHashChangelogUpdates = await updatePackages(
-		gitWorkingDirectoryPath,
-		minimumVersionBump,
-		releaseType,
-		abortMessage
-	);
+	let pluginReleaseBranch;
+	if ( [ 'latest', 'next' ].includes( config.releaseType ) ) {
+		pluginReleaseBranch = await findPluginReleaseBranchName(
+			config.gitWorkingDirectoryPath
+		);
+		await runNpmReleaseBranchSyncStep( pluginReleaseBranch, config );
+	} else {
+		await checkoutNpmReleaseBranch( config );
+	}
 
-	await runPushGitChangesStep(
-		gitWorkingDirectoryPath,
-		releaseBranch,
-		`Aborting! Make sure to push changes applied to WordPress release branch "${ releaseBranch }" manually.`
-	);
+	const commitHashChangelogUpdates = await updatePackages( config );
 
-	const commitHashNpmPublish = await publishPackagesToNpm(
-		gitWorkingDirectoryPath,
-		minimumVersionBump,
-		releaseType
-	);
+	const commitHashNpmPublish = await publishPackagesToNpm( config );
 
-	await backportCommitsToTrunk(
-		gitWorkingDirectoryPath,
-		releaseType,
-		[ commitHashChangelogUpdates, commitHashNpmPublish ].filter( Boolean )
-	);
+	if ( [ 'latest', 'bugfix' ].includes( config.releaseType ) ) {
+		const commits = [
+			commitHashChangelogUpdates,
+			commitHashNpmPublish,
+		].filter( Boolean );
+		await backportCommitsToBranch( 'trunk', commits, config );
+
+		if ( config.releaseType === 'latest' && pluginReleaseBranch ) {
+			await backportCommitsToBranch(
```

Comment:
The part for the plugin release branch didn't work correctly. There were merge conflicts in every updated file. I need to look closer into it.